packages/api-v2/src/student/student.controller.ts

```diff
@@ -102,7 +103,8 @@ export class StudentController {
     return student;
   }
 
-  @UseGuards(JwtAuthGuard)
+  // TODO: Should users need to be confirmed to delete their account
```

Comment:
interesting question, for now yes... since the only way to achieve user deletion will be through the home page which won't be available till email is confirmed.